pkg/openstack/tasks/servers.go

```diff
@@ -0,0 +1,258 @@
+}
+
+// // HandleCollectServersTask handles the task for collecting OpenStack Servers.
+func HandleCollectServersTask(ctx context.Context, t *asynq.Task) error {
+	// If we were called without a payload, then we enqueue tasks for
+	// collecting OpenStack Servers from all known projects.
+	data := t.Payload()
+	if data == nil {
+		return enqueueCollectServers(ctx)
+	}
+
+	var payload CollectServersPayload
+	if err := asynqutils.Unmarshal(data, &payload); err != nil {
+		return asynqutils.SkipRetry(err)
+	}
+
+	return collectServers(ctx, payload)
+}
+
+// enqueueCollectServers enqueues tasks for collecting OpenStack Servers from
+// all configured OpenStack projects by creating a payload with the respective
+// project ID.
+func enqueueCollectServers(ctx context.Context) error {
+	logger := asynqutils.GetLogger(ctx)
+
+	if openstackclients.ComputeClientset.Length() == 0 {
+		logger.Warn("no OpenStack compute clients found")
+		return nil
+	}
+
+	err := openstackclients.ComputeClientset.Range(func(projectID string, client openstackclients.Client[*gophercloud.ServiceClient]) error {
+		payload := CollectServersPayload{
+			ProjectID: projectID,
+		}
+		data, err := json.Marshal(payload)
+		if err != nil {
+			logger.Error(
+				"failed to marshal payload for OpenStack servers",
+				"project_id", projectID,
+				"reason", err,
+			)
+			return err
+		}
+
+		task := asynq.NewTask(TaskCollectServers, data)
+		info, err := asynqclient.Client.Enqueue(task)
+		if err != nil {
+			logger.Error(
+				"failed to enqueue task",
+				"type", task.Type(),
+				"project_id", projectID,
+				"reason", err,
+			)
+			return err
+		}
+
+		logger.Info(
+			"enqueued task",
+			"type", task.Type(),
+			"id", info.ID,
+			"queue", info.Queue,
+			"project_id", projectID,
+		)
+
+		return nil
+	})
+
+	if err != nil {
+		logger.Error(
+			"could not get project clients",
+			"reason", err,
+		)
+	}
+
+	return nil
+}
+
+// collectServer collects the OpenStack servers from the specified project,
+// using the client associated with the project ID in the given payload.
+func collectServers(ctx context.Context, payload CollectServersPayload) error {
+	logger := asynqutils.GetLogger(ctx)
+
+	projectClients := make([]*openstackclients.Client[*gophercloud.ServiceClient], 0)
+	err := openstackclients.ComputeClientset.Range(func(projectID string, client openstackclients.Client[*gophercloud.ServiceClient]) error {
+		if projectID == payload.ProjectID {
+			projectClients = append(projectClients, &client)
+		}
+		return nil
+	})
+
+	if err != nil {
+		logger.Error(
+			"could not get project clients",
+			"project_id", payload.ProjectID,
+			"reason", err,
+		)
+		return err
+	}
+
+	logger.Info(
+		"project clients found",
+		"project", payload.ProjectID,
+		"clients", len(projectClients),
+	)
+
+	if len(projectClients) == 0 {
+		return asynqutils.SkipRetry(ClientNotFound(payload.ProjectID))
+	}
+
+	for _, client := range projectClients {
```

Comment:
Please see previous comment about having multiple clients for the same project in the registry.